dive_core/command_hierarchy.cpp

```diff
@@ -105,6 +108,25 @@ uint64_t Topology::GetChildNodeIndex(uint64_t node_index, uint64_t child_index)
     DIVE_ASSERT(child_list_index < m_children_list.size());
     return m_children_list[child_list_index];
 }
+
+//--------------------------------------------------------------------------------------------------
+uint64_t Topology::GetNextNodeIndex(uint64_t node_index) const
+{
+    uint64_t num_children = GetNumChildren(node_index);
+    if (num_children > 0)
+        return GetChildNodeIndex(node_index, 0);
+    while (true)
+    {
+        if (node_index == kRootNodeIndex)
+            return UINT64_MAX;
+        uint64_t parent_node_index = GetParentNodeIndex(node_index);
+        uint64_t sibling_index = GetChildIndex(node_index) + 1;
```

Comment:
a bit confused with the logic here, if the code reaches here, that means GetNumChildren(node_index) <= 0, so the current node (with node_index) should not have any child?

Reply:
I didn't modify this function at all for this PR, and in fact this GetNextNodeIndex() function isn't being used at all. The diff is only making this part show up because it's being weird.

I think the logic here is that if a node doesn't have children, then it's trying to search for "next node index" in sibling branches. We should probably remove this in a future PR since this isn't being used anymore.